pallets/xyk/src/tests.rs

```diff
@@ -2200,3 +2158,103 @@ fn get_fee_should_work() {
 		assert_eq!(fee, (2, 1_000));
 	});
 }
+
+#[test]
+fn cannot_drain_pool() {
```

Comment:
this test is actually a very, very specific case... I'd suggest something more like "fn share_ratio_calculations_are_correct" even though that's still pretty generic, it's more targetted. This test is definitely not sufficient to guarantee that the pool cannot be drained.